tests/group_functions/group_scan.h

```diff
@@ -20,11 +20,149 @@
+    sycl::buffer<U, 1> res_i_sycl(res_i.data(), sycl::range<1>(size));
+    sycl::buffer<bool, 1> end_e_sycl(&res_e_end, sycl::range<1>(1));
+    sycl::buffer<bool, 1> end_i_sycl(&res_i_end, sycl::range<1>(1));
+    sycl::buffer<bool, 1> ret_type_e_sycl(&ret_type_e, sycl::range<1>(1));
+    sycl::buffer<bool, 1> ret_type_i_sycl(&ret_type_i, sycl::range<1>(1));
+
+    queue
+        .submit([&](sycl::handler& cgh) {
+          auto v_acc =
+              v_sycl.template get_access<sycl::access::mode::read>(cgh);
+          auto res_e_acc =
+              res_e_sycl.template get_access<sycl::access::mode::read_write>(
+                  cgh);
+          auto res_i_acc =
+              res_i_sycl.template get_access<sycl::access::mode::read_write>(
+                  cgh);
+          auto end_e_acc =
+              end_e_sycl.template get_access<sycl::access::mode::read_write>(
+                  cgh);
+          auto end_i_acc =
+              end_i_sycl.template get_access<sycl::access::mode::read_write>(
+                  cgh);
+          auto ret_type_e_acc =
+              ret_type_e_sycl
+                  .template get_access<sycl::access::mode::read_write>(cgh);
+          auto ret_type_i_acc =
+              ret_type_i_sycl
+                  .template get_access<sycl::access::mode::read_write>(cgh);
+
+          cgh.parallel_for<
+              joint_scan_group_kernel<D, T, U, Group, with_init, I, OpT>>(
+              executionRange, [=](sycl::nd_item<D> item) {
+                Group group = get_group<Group>(item);
+
+                T* v_begin = v_acc.get_pointer();
+                T* v_end = v_begin + v_acc.size();
+                U* r_e_begin = res_e_acc.get_pointer();
+                U* r_i_begin = res_i_acc.get_pointer();
+
+                auto r_e_end = joint_exclusive_scan_helper<with_init, I>(
+                    group, v_begin, v_end, r_e_begin, op);
+                ret_type_e_acc[0] = std::is_same_v<U*, decltype(r_e_end)>;
+
+                auto r_i_end = joint_inclusive_scan_helper<with_init, I>(
+                    group, v_begin, v_end, r_i_begin, op);
+                ret_type_i_acc[0] = std::is_same_v<U*, decltype(r_i_end)>;
+
+                end_e_acc[0] = (r_e_begin + res_e_acc.size() == r_e_end);
+                end_i_acc[0] = (r_i_begin + res_i_acc.size() == r_i_end);
+              });
+        })
+        .wait_and_throw();
+  }
+  CHECK(res_e_end);
+  CHECK(res_i_end);
+  CHECK(ret_type_e);
+  CHECK(ret_type_i);
+  std::vector<U> reference_e(size, U(-1));
+  std::vector<U> reference_i(size, U(-1));
+
+// FIXME: known_identity is not implemented in hipSYCL yet
+#if SYCL_CTS_COMPILING_WITH_HIPSYCL
+  I init_value = (with_init) ? I(init)
+                 : (std::is_same_v<OpT, sycl::plus<U>>)
+                     ? I{}
+                     : std::numeric_limits<I>::lowest();
+#else
+  I init_value = (with_init) ? I(init) : sycl::known_identity<OpT, I>::value;
+#endif
```

Comment:
This is another place where I think we should remove `#if SYCL_CTS_COMPILING_WITH_HIPSYCL`.
It's reasonable to require `sycl::known_identity` support for all implementations.

Reply:
Removed workaround for known_identity and disable tests for hipSYCL.